crates/router/src/core/payment_methods/surcharge_decision_configs.rs

```diff
@@ -317,13 +334,98 @@ pub async fn perform_surcharge_decision_management_for_saved_cards(
                     .change_context(ConfigError::DslExecutionError)
             })
             .transpose()?;
+
+        backend_input.payment_method.card_network = card_network;
+
+        let surcharge_details = surcharge_source
+            .generate_surcharge_details_and_populate_surcharge_metadata(
+                &backend_input,
+                payment_attempt,
+                (
+                    &mut surcharge_metadata,
+                    types::SurchargeKey::Token(payment_token),
+                ),
+            )?;
+        customer_payment_method.surcharge_details = surcharge_details
+            .map(|surcharge_details| {
+                SurchargeDetailsResponse::foreign_try_from((&surcharge_details, payment_attempt))
+                    .change_context(ConfigError::DslParsingError)
+            })
+            .transpose()?;
+    }
+    Ok(surcharge_metadata)
+}
+
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+pub async fn perform_surcharge_decision_management_for_saved_cards(
```

Comment:
what us surcharge_decision_management. write a comment explaining that or change the function name to give more specific meaning